crates/rover-client/src/error.rs

```diff
@@ -20,12 +20,34 @@ pub enum RoverClientError {
     InvalidJSON(#[from] serde_json::Error),
 
     /// Encountered an error handling the received response.
```

Comment:
```suggestion
    /// Generic one-off error without a defined error type
```

Reply:
I think having a comment about not being able to handle a response makes a bit more sense to me (especially given how we are using this error type).